File: Latihan2_1/Hardware.java
// class Hardware adalah child dari class Product dan parent dari class Memory
public class Hardware extends Product {
    private String brand;
    private String model;
    
    // constructor
    public Hardware(String brand, String model) {
        this.brand = brand;
        this.model = model;
    }

    // constructor kosong
    public Hardware() {
    }

    // set memory brand
    public void set_brand(String brand) {
        this.brand = brand;
    }
    // get memory brand
    public String get_brand() {
        return brand;
    }

    // set memory model
    public void set_model(String model) {
        this.model = model;
    }
    // get memory model
    public String get_model() {
        return model;
    }
}

File: Latihan2_2/php/Hardware.php
<?php

// class Hardware adalah child dari class Product dan parent dari class Memory
class Hardware extends Product
{
    private $brand = "";
    private $model = "";

    //constructor
    function __construct(){
    }

    // set memory brand
    function set_brand($brand)
    {
        $this->brand = $brand;
    }
    // get memory brand
    function get_brand()
    {
        return $this->brand;
    }

    // set memory model
    function set_model($model)
    {
        $this->model = $model;
    }
    // get memory model
    function get_model()
    {
        return $this->model;
    }

    // destructor
    function __destruct(){
    }
}

?>